### DISCUSSION TITLE
How does this project differ from https://github.com/ahmetoner/whisper-asr-webservice/ ?

### DISCUSSION THREAD

**Discussion by rgreen83** [McCloudS/subgen#116] [Q&A] [answered]
I use https://github.com/ahmetoner/whisper-asr-webservice/ currently as a docker container and with Bazarr both hosted on an Unraid server. It seems to work ok but the provider times out in bazarr very frequently. I can't tell if this project is significantly different or just a similar alternative? Since there isn't a published template for this docker it is pretty difficult to deploy as a container in Unraid so its not easy to just try it out and see for myself.

**Comment by McCloudS** [comment, marked as answer]
It's the same underlying idea, just Subgen uses stable-ts and other tweaks to clean up the output to make it more it a more natural subtitle output.  Subgen loosely follows Netflix's subtitling rules (https://partnerhelp.netflixstudios.com/hc/en-us/articles/215758617-Timed-Text-Style-Guide-General-Requirements).  Additionally it, provides more verbose logging that whisper-asr doesn't, which is helpful for troubleshooting.  It also has way more configurable options and overrides, to include interfacing with Plex, Emby, Jellyfin, and Tautulli if someone isn't using Bazarr.

Big options that Subgen has now (that ASR doesn't):

- Uses Stable-ts for more reliable timestamping
- Plex, Emby, Jellyfin, and Tautulli integrations (not relevant to Bazarr users)
- Control threads and concurrent transcriptions
- Force a detected language or override the detected language
- Web interface for configuration
- Endpoint to manually run transcriptions if desired (/batch)
- Clears (V)RAM when not in use
- Can append a message at the end of the subtitle saying it was generated by Subgen/AI
- Can 'monitor' a folder for files needing transcription
- Can take custom model prompts
- Can create LRC files for music
- Can take a custom regroup command to format the subtitle output as you desire
- Can extend the default detect_language to any length
- Can accept KWARGS which allow you to customize virtually any input from the Whisper model or transcription function

Whisper-ASR uses faster-whisper and that is it, no customization, no optimization for subtitle output.  Subgen does the same, with just a bit more.

Subgen is easy to use with unraid (which is my daily driver).  See https://github.com/McCloudS/subgen/issues/37 for an example.  

Subgen likely won't fix your timeout issue (without knowing what the error is).  If you're using CPU only, you need to bump your transcription timeout in Bazarr to account for that.

**Reply by rgreen83** [reply]
Thank you so much for the in depth reply! I am gonna follow that link and look forward to trying this out!
